### DISCUSSION TITLE
Rivet-switch Version

### DISCUSSION THREAD

**Discussion by 0jacksonl** [indiana-university/rivet-source#439] [Questions] [answered]
Greetings,
 
We're getting the following browser messages in ACM UI's, and wondered if
or when the Switch component will be updated so that it doesn't happen:
 
react-dom.development.js?61bb:88 Warning: componentWillReceiveProps has been renamed, and is not recommended for use. See https://fb.me/react-unsafe-component-lifecycles for details.
 
* Move data fetching code or side effects to componentDidUpdate.
* If you're updating state whenever props change, refactor your code to use memoization techniques or move it to static getDerivedStateFromProps. Learn more at: https://fb.me/react-derived-state
* Rename componentWillReceiveProps to UNSAFE_componentWillReceiveProps to suppress this warning in non-strict mode. In React 17.x, only the UNSAFE_ name will work. To rename all deprecated lifecycles to their new names, you can run `npx react-codemod rename-unsafe-lifecycles` in your project source folder.
 
Please update the following components: Switch

Is updating Switch anywhere on the horizon?

Thanks,

JLJ

**Comment by levimcg** [comment, marked as answer]
@0jacksonl I was able to respond to your email separately, but did want to post my answers here just in case anyone else runs into the same errors.

---
 It looks like you might be using the `rivet-react` package which is a separate community-maintained implementation of the core Rivet styles in React components. The best place to bring that up would be as an issue on that repository here: https://github.com/indiana-university/rivet-react/issues.
 
There wouldn’t be any internals in the main Rivet JavaScript codebase (here in this repo) that would throw React-specific errors, which is what it is looks like is happening with one of React's deprecated class component lifecycle methods.
